### ISSUE TITLE
`option_env_unwrap` doesn't warn if the env variable isn't set

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust-clippy#10742]
### Summary

`option_env_unwrap` only lints if the variable is currently set in the environment. This makes it hard to reproduce clippy output between machines; for example, CI often sets test variables that aren't set locally. Clippy should warn about these consistently, even if the variables aren't set in the environment.

Here's a slightly more realistic example of why it would be useful to warn:
```rust
#[test]
fn test_simple() {
  // ...
}

#[test]
#[ignored = "needs credentials"]
fn test_extended() {
  let credentials = option_env!("CREDENTIALS").unwrap();
  // ...
}
```

### Lint Name

option_env_unwrap

### Reproducer

I tried this code:

```rust
#[allow(unused)]
fn foo() {
    let x = option_env!("CARGO").unwrap();
    let y = option_env!("Y").unwrap();
}
```

I expected to see this happen: Both `CARGO` and `Y` have the lint emitted.

Instead, this happened: Only `CARGO` had the lint emitted.

### Version

```text
Clippy 0.1.70 (2023-05-03 473f916)
```
